src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLGuid.cs

```diff
@@ -292,7 +281,7 @@ public int CompareTo(SqlGuid value)
             (this == other).Value;
 
         // For hashing purpose
-        public override int GetHashCode() => IsNull ? 0 : Value.GetHashCode();
+        public override int GetHashCode() => _value.GetHashCode();
```

Comment:
I think you could also change `Equals` above to:

```diff
        public bool Equals(SqlGuid other) =>
-            other.IsNull || IsNull ? other.IsNull && IsNull :
-            (this == other).Value;
+            _value == other._value;
```

and then the operator at line 152 to:

```diff
        public static SqlBoolean operator ==(SqlGuid x, SqlGuid y)
        {
-            return (x.IsNull || y.IsNull) ? SqlBoolean.Null : new SqlBoolean(Compare(x, y) == EComparison.EQ);
+            return (x.IsNull || y.IsNull) ? SqlBoolean.Null : new SqlBoolean(x.Equals(y));
        }
```

This avoids the custom byte-wise compare for each